File: sentimental-cash/cash.py
from cs50 import get_float


def main():
    # ask how many dollars the customer is owed.
    dollars = get_dollars()

    # calculate number of quarters
    quarters = calculate_quarters(dollars)
    dollars = dollars - quarters * 0.25

    # calculate number of dimes
    dimes = calculate_dimes(dollars)
    dollars = round(dollars - dimes * 0.10, 2)

    # calculate number of nickles
    nickels = calculate_nickels(dollars)
    dollars = round(dollars - nickels * 0.05, 2)

    # calculate number of pennies
    pennies = calculate_pennies(dollars)
    dollars = round(dollars - pennies * 0.01, 2)

    # sum coins
    coins = quarters + dimes + nickels + pennies

    # print total number of coins
    print(coins)


def get_dollars():
    while True:
        change = get_float("Change owed:\n")
        if change > 0:
            return change


def calculate_quarters(dollars):
    q = 0
    d = dollars
    value = float(0.25)

    while d >= value:
        d = d - value
        q += 1

    return q


def calculate_dimes(dollars):
    di = 0
    d = dollars
    value = float(0.10)

    while d >= value:
        d = d - value
        di += 1

    return di


def calculate_nickels(dollars):
    n = 0
    d = dollars
    value = float(0.05)

    while d >= value:
        d = round(d - value, 2)
        n += 1

    return n


def calculate_pennies(dollars):
    p = 0
    d = dollars
    value = float(0.01)

    while d >= value:
        d = d - value
        p += 1

    return p


main()

File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

int get_cents(void)
{
    // TODO
    int change;
    do
    {
        change = get_int("Change owed:\n");
    }
    while (change < 0);
    return change;
}

int calculate_quarters(int cents)
{
    // TODO
    int q;
    int c = cents;

    for (q = 0; c >= 25; q++)
    {
        c = c - 25;
    }

    return q;

}

int calculate_dimes(int cents)
{
    // TODO
    int d;
    int c = cents;

    for (d = 0; c >= 10; d++)
    {
        c = c - 10;
    }

    return d;

}

int calculate_nickels(int cents)
{
    // TODO
    int n;
    int c = cents;

    for (n = 0; c >= 5; n++)
    {
        c = c - 5;
    }

    return n;
}

int calculate_pennies(int cents)
{
    // TODO
    int p;
    int c = cents;

    for (p = 0; c >= 1; p++)
    {
        c = c - 1;
    }

    return p;

}
